modules/contact/src/constraints/ComputeDynamicFrictionalForceLMMechanicalContact.C

```diff
@@ -107,28 +107,9 @@ ComputeDynamicFrictionalForceLMMechanicalContact::computeQpProperties()
   // Compute the value of _qp_gap
   ComputeDynamicWeightedGapLMMechanicalContact::computeQpProperties();
 
-  ADRealVectorValue relative_velocity;
-
-  if (_3d)
-    relative_velocity = {_secondary_x_dot[_qp] - _primary_x_dot[_qp],
-                         _secondary_y_dot[_qp] - _primary_y_dot[_qp],
-                         (*_secondary_z_dot)[_qp] - (*_primary_z_dot)[_qp]};
-  else
-    relative_velocity = {_secondary_x_dot[_qp] - _primary_x_dot[_qp],
-                         _secondary_y_dot[_qp] - _primary_y_dot[_qp],
-                         0.0};
-
-  // Add derivative information
-  relative_velocity(0).derivatives() =
-      _secondary_x_dot[_qp].derivatives() - _primary_x_dot[_qp].derivatives();
-  relative_velocity(1).derivatives() =
-      _secondary_y_dot[_qp].derivatives() - _primary_y_dot[_qp].derivatives();
-
-  if (_3d)
-    relative_velocity(2).derivatives() =
-        (*_secondary_z_dot)[_qp].derivatives() - (*_primary_z_dot)[_qp].derivatives();
-
-  _qp_tangential_velocity_nodal = relative_velocity * (_JxW_msm[_qp] * _coord[_qp]);
+  // It appears that the relative velocity between weighted gap and this class have a sign
```

Comment:
Next time I touch this, I'll look into making this more consistent and avoid this ugly negative sign here.